packages/analytics-browser/src/browser-client.ts

```diff
@@ -86,7 +87,7 @@ export class AmplitudeBrowser extends AmplitudeCore implements BrowserClient {
 
     // Step 4: Install plugins
     // Do not track any events before this
-    await this.add(new Destination()).promise;
+    await this.add(new Destination(this.config)).promise;
```

Comment:
all plugins receive a config object through setup method, probably not needed

Reply:
[The setup method](https://github.com/amplitude/Amplitude-TypeScript/blob/ad928eff9556a3456667593e9acae9d47c5bf1b9/packages/analytics-core/src/timeline.ts#L25-L30) in `CoreClient.register()` accepts an `CoreConfig` type without a `DiagnosticProvider`. 

I only add a `DiagnosticProvider` to `BrowserConfig`.

As this is internal, I don't see any harm to pass a config here. Otherwise, have to create a `BrowserTimeline` to overrider the register method in Core.

Reply:
Pretty sure that the browser specific config would be present in the config object passed to setup. We might just need to adjust the type definition of the plugin